src/containers/AccountConnection.jsx

```diff
@@ -123,75 +130,78 @@ class AccountConnection extends Component {
       .catch(error => this.handleError(error))
   }
 
-  handleCreateSuccess (account) {
-    const messages = [{message: 'account.message.success.config'}]
+  handleDeleteSuccess () {
+    this.setState({
+      submitting: false,
+      deleting: false,
+      error: null,
+      success: null // exception for the delete success which uses alerts
+    })
 
-    if (account.folderPath) {
-      messages.push({message: 'account.message.details', params: {folder: account.folderPath}})
-    }
+    this.props.alertSuccess([{message: 'account.message.success.delete'}])
+  }
 
-    if (this.props.connector.additionnalSuccessMessage) {
-      messages.push(
-        {
-          message: this.props.connector.additionnalSuccessMessage.message || ''
-        }
-      )
-    }
-    this.handleSuccess(account, messages)
+  handleCreateSuccess () {
+    const { t } = this.context
+    const messages = [t('account.message.success.connect', {name: this.props.connector.name})]
+    this.handleSuccess(SUCCESS_TYPES.CREATE, messages)
   }
 
-  handleUpdateSuccess (account) {
-    const messages = [{message: 'account update success'}]
-    if (this.props.connector.additionnalSuccessMessage) {
-      messages.push(
-        {
-          message: this.props.connector.additionnalSuccessMessage.message || ''
-        }
-      )
-    }
-    this.handleSuccess(account, messages)
+  handleSuccessTimedOut () {
+    const { t } = this.context
+    const messages = [t('account.message.success.connect', {name: this.props.connector.name})]
+    this.handleSuccess(SUCCESS_TYPES.TIMEDOUT, messages)
+  }
+
+  handleUpdateSuccess () {
+    const { t } = this.context
+    const messages = [t('account.message.success.update', {name: this.props.connector.name})]
+    this.handleSuccess(SUCCESS_TYPES.UPDATE, messages)
   }
 
-  handleSuccess (account, messages = []) {
+  handleSuccess (type, messages = []) {
+    const { t } = this.context
+    if (this.props.connector.additionnalSuccessMessage && this.props.connector.additionnalSuccessMessage.message) {
+      messages.push(t(this.props.connector.additionnalSuccessMessage.message))
+    }
     this.setState({
       submitting: false,
       deleting: false,
-      error: undefined,
-      credentialsError: undefined
+      error: null,
+      success: {
+        type,
+        messages
+      }
     })
-
-    this.props.onSuccess(account, messages)
   }
 
   handleError (error) {
-    const stateUpdate = {
-      submitting: false,
-      deleting: false
-    }
-
+    // timed out exception
     if (error.message === ACCOUNT_ERRORS.SUCCESS_TIMEDOUT) {
-      stateUpdate.isSuccessTimedOut = true
-      stateUpdate.error = null
-      stateUpdate.credentialsError = null
-    } else if (error.message === ACCOUNT_ERRORS.LOGIN_FAILED) {
-      stateUpdate.credentialsError = error
-    } else {
-      stateUpdate.error = error
+      // only timed out "error" return will have an account property
```

Comment:
Wouldn’t `timed out "error" have an account property` be a more durable comment ?